sklearn/feature_extraction/text.py

```diff
@@ -1054,6 +1054,26 @@ def idf_(self):
 class TfidfVectorizer(CountVectorizer):
     """Convert a collection of raw documents to a matrix of TF-IDF features.
 
+    Tf means term-frequency while tf-idf means term-frequency times inverse
+    document-frequency. This is a common term weighting scheme in information
+    retrieval, that has also found good use in document classification.
+
+    The goal of using tf-idf instead of the raw frequencies of occurrence of a
+    token in a given document is to scale down the impact of tokens that occur
+    very frequently in a given corpus and that are hence empirically less
+    informative than features that occur in a small fraction of the training
+    corpus.
+
+    The actual formula used for tf-idf is tf * (idf + 1) = tf + tf * idf,
```

Comment:
Interesting. I never knew we did this. Can @jnothman shed some light on if this is a common practise? It does not show up here (https://en.wikipedia.org/wiki/Tf%E2%80%93idf#Inverse_document_frequency_2) if I am right.